src/features/code_actions.zig

```diff
@@ -366,6 +385,281 @@ fn handleVariableNeverMutated(builder: *Builder, actions: *std.ArrayListUnmanage
+        const lhs_kind = @intFromEnum(lhs.getKind());
+        const rhs_kind = @intFromEnum(rhs.getKind());
+        if (rhs_kind <= @intFromEnum(Kind.build_options)) return false;
+        return lhs_kind != rhs_kind;
+    }
+
+    pub fn getSourceStartIndex(self: ImportDecl, tree: Ast) usize {
+        return offsets.tokenToIndex(tree, self.first_comment_token orelse tree.firstToken(self.var_decl));
+    }
+
+    pub fn getSourceEndIndex(self: ImportDecl, tree: Ast, include_line_break: bool) usize {
+        const token_tags = tree.tokens.items(.tag);
+
+        var last_token = ast.lastToken(tree, self.var_decl);
+        if (last_token + 1 < tree.tokens.len - 1 and token_tags[last_token + 1] == .semicolon) {
+            last_token += 1;
+        }
+
+        var end = offsets.tokenToLoc(tree, last_token).end;
+        if (!include_line_break) return end;
+        while (end < tree.source.len) {
+            switch (tree.source[end]) {
+                ' ', '\t', '\n' => end += 1,
+                else => break,
+            }
+        }
+        return end;
+    }
+
+    /// similar to `offsets.nodeToLoc` but will also include preceding comments and postfix semicolon and line break
+    pub fn getLoc(self: ImportDecl, tree: Ast, include_line_break: bool) offsets.Loc {
+        return .{
+            .start = self.getSourceStartIndex(tree),
+            .end = self.getSourceEndIndex(tree, include_line_break),
+        };
+    }
+};
+
+pub fn getImportsDecls(builder: *Builder, allocator: std.mem.Allocator) error{OutOfMemory}!std.ArrayListUnmanaged(ImportDecl) {
+    const tree = builder.handle.tree;
+    var imports = std.ArrayListUnmanaged(ImportDecl){};
+    errdefer imports.deinit(allocator);
+
+    const node_tags = tree.nodes.items(.tag);
+    const node_data = tree.nodes.items(.data);
+    const node_tokens = tree.nodes.items(.main_token);
+
+    // iterate until no more imports are found
+    var updated = true;
+    while (updated) {
+        updated = false;
+        next_decl: for (tree.rootDecls()) |node| {
+            if (node_tags[node] != .simple_var_decl) continue;
+
+            // check if we already have this import
+            for (imports.items) |import_decl| {
+                if (import_decl.var_decl == node) continue :next_decl;
+            }
+            const var_decl = tree.simpleVarDecl(node);
+            const base_token = var_decl.ast.init_node;
+
+            var inode = base_token;
+            const import: ImportDecl = found_decl: while (true) {
+                const token = node_tokens[inode];
+                switch (node_tags[inode]) {
+                    .builtin_call_two, .builtin_call_two_comma => {
+                        // @import("string") case
+                        const init_node = inode;
+                        const call_name = offsets.tokenToSlice(tree, token);
+                        if (!std.mem.eql(u8, call_name, "@import")) continue;
```

Comment:
This `continue` will always be an infinite loop. Nothing is being modified on each iteration so you will hit this continue again and again.
Example Test:
```zig
test "organize imports - @embedFile" {
    try testDiagnostic(
        \\const foo = @embedFile("foo.zig");
    , "infinite loop go brrrrrrrr");
}
```
```suggestion
                        if (!std.mem.eql(u8, call_name, "@import")) break;
```